generators/generator-base.js

```diff
@@ -1320,29 +1320,15 @@ module.exports = class extends PrivateBase {
 
     /**
      * executes a Git command using shellJS
-     * gitExec(args [, options ], callback)
+     * gitExec(args [, options] [, callback])
      *
      * @param {string|array} args - can be an array of arguments or a string command
      * @param {object} options[optional] - takes any of child process options
-     * @param {function} callback - a callback function to be called once process complete, The call back will receive code, stdout and stderr
+     * @param {function} callback[optional] - a callback function to be called once process complete, The call back will receive code, stdout and stderr
+     * @return {object} when in synchronous mode, this returns a ShellString. Otherwise, this returns the child process object.
      */
     gitExec(args, options, callback) {
-        callback = arguments[arguments.length - 1]; // eslint-disable-line prefer-rest-params
-        if (arguments.length < 3) {
-            options = {};
-        }
-        if (options.async === undefined) options.async = true;
-        if (options.silent === undefined) options.silent = true;
-        if (options.trace === undefined) options.trace = true;
-
-        if (!Array.isArray(args)) {
-            args = [args];
-        }
-        const command = `git ${args.join(' ')}`;
-        if (options.trace) {
-            this.info(command);
-        }
-        shelljs.exec(command, options, callback);
+        return jhipsterUtils.gitExec(args, options, callback);
```

Comment:
This one is in generator-base.
We should keep compatibility.
```
if (typeof options === 'function') {
  callback = options;
}
```

Reply:
Add that to utils.gitExec.

Reply:
> This one is in generator-base.
> We should keep compatibility.
> 
> ```
> if (typeof options === 'function') {
>   callback = options;
> }
> ```

Sorry, but I can't find that in generator-base. Are you sure its there?

Reply:
generator-base.js should be api stable.
https://github.com/jhipster/generator-jhipster/blob/448d48ec9df570ee4c68f46050e23c0c936b49b7/generators/generator-base.js#L43-L49

https://github.com/jhipster/generator-jhipster/blob/448d48ec9df570ee4c68f46050e23c0c936b49b7/generators/generator-base.js#L1330-L1333
Was removed. It allowed `gitExec('...', callback)`.
So we should add 
```
if (typeof options === 'function') {
  callback = options;
}
```
To fix api compatibility.
This could be added to gitExec at generator-base, but it will be better to add to gitExec at utils.

Reply:
You are right. Did not realize although you actually pointed it out in your first comment as well. Thought I have moved the code verbatim. Anyways, fixed it as per your suggestion